drud/ddev
# Add global custom commands, fixes #2054

## The Problem/Issue/Bug:

Allow global custom commands per #2054
- [x] Make sure command names are unique. 
- [x] Project-level command should overwrite global.
- [x] Try to make dotfile .ddev/commands/.web, etc not break everything. Move those to .commands maybe
- [x] .gitignore in global .ddev

## How this PR Solves The Problem:

Custom commands can now be placed in ~/.ddev/comands exactly the same as they're placed in the project's .ddev/commands directory.

## Manual Testing Instructions:

See the readme in ~/.ddev/commands/* to add a new command. Inside a project you should see it show up when you type "ddev"

## Automated Testing Overview:

TestCustomCommands was enhanced to make sure global commands work and are overridden by project-level commands.

## Related Issue Link(s):

Fixes #2054 

## Release/Deployment notes:
